src/SingleLinkedList.h

```diff
@@ -145,6 +145,7 @@ void SingleLinkedList<T>::add(int index, T* element) {
 		add(element);
 	} else if(index == 0){
 		Node<T>* newNode = new Node<T>(element, head);
+		newNode->next = head->next;
 		head = newNode;
 		size++;
```

Comment:
I'm so sorry DJ Must have been half asleep last night. I think what you had before was correct. What I think we just did was remove the first element in the list and add this one instead. So the code below fixes it to what should happen just adding the node.

```c++
Node<T>* newNode = new Node<T>(element, head);
newNode->next = head;
head = newNode;
size++;
```